nameko/cli/main.py

```diff
@@ -45,8 +45,11 @@ def _replace_env_var(match):
 
 
 def _env_var_constructor(loader, node):
-    value = loader.construct_scalar(node)
-    return ENV_VAR_MATCHER.sub(_replace_env_var, value)
+    raw_value = loader.construct_scalar(node)
+    value = ENV_VAR_MATCHER.sub(_replace_env_var, raw_value)
+    new_tag = loader.resolve(yaml.ScalarNode, value, (True, False))
```

Comment:
This (True, False) magic - only way to say pyyaml: try resolve node by content and do not return **string** in all cases. Via hinter from **yaml.add_implicit_resolver('!env_var', IMPLICIT_ENV_VAR_MATCHER)** any yaml node with ${...} routed to _env_var_constructor() and avoid pyyaml resolver. So when u use yaml
`
a: True
b: ${env_b}
`
without this fix - always boolean **a** and string **b**, even if env_b == "True"